elastic/elastic-agent
# [OTel]: enable persistence by default in our OTel distribution

- Enhancement

## What does this PR do?
- By default, the filelog receiver stores offsets in memory rather than on disk. Since users might forget to enable on-disk storage themselves, this PR enables persistence by default in our sample configurations.

## Why is it important?
- If the collector is restarted, there could either be data duplication (if it starts re-reading files from the beginning) or data loss (if it starts reading files from the end and new data has been written to these files during the restart).

## Checklist




## Disruptive User Impact



## How to test this PR locally

- Pull from this branch.
- Run `elastic-agent otel` 
- Try ingesting some data
- Shutdown the otel command
- Rerun the `elastic-agent otel` 
- Write new data
- You shouldn't see duplicated data, as offsets will be stored on disk.

## Related issues


-

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
